File: two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& a, int t) {
        unordered_map<int,int> m;
        for(int i=0;i<a.size();i++) {
            if(m[t-a[i]]) return {m[t-a[i]]-1,i};
            m[a[i]]=i+1;
        }
        return {};
    }
};

File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int temp[] = new int[2];
        HashMap<Integer, Integer> hmap = new HashMap<Integer, Integer>();
        for(int i=0;i<nums.length;i++) {
            hmap.put(nums[i], i);
        }
        for(int i=0;i<nums.length;i++) {
            if(hmap.containsKey(target-nums[i])) {
                temp[0]=i; temp[1]=hmap.get(target-nums[i]);
                if(temp[0]!=temp[1]) return temp;
            }
        }
        return temp;

    }
}